### PR TITLE
Fix some function DATE type priority

### PR DESCRIPTION
**Pull request by kangkaisen**
1. Fix the bug introduced by https://github.com/apache/incubator-doris/pull/2947.  
The following sql result is 0000, which is wrong. The result should be 1601
```
select date_format('2020-02-19 16:01:12','%H%i');
```

2. Add constant Express plan test, ensure the FE constant Express compute result is right.

3. Remove the `castToInt ` function in `FEFunctions`, which is duplicated with `CastExpr::getResultValue`

4. Implement `getNodeExplainString` method for `UnionNode`

### CODE REVIEW

File: fe/src/test/java/org/apache/doris/utframe/ConstantExpressTest.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
+package org.apache.doris.utframe;
```
Comment (imay) [inline fe/src/test/java/org/apache/doris/utframe/ConstantExpressTest.java]: license header

File: fe/src/main/java/org/apache/doris/planner/SingleNodePlanner.java [hunk 2]
```diff
@@ -802,7 +802,6 @@ private PlanNode createConstantSelectPlan(SelectStmt selectStmt, Analyzer analyz
         for (int i = 0; i < resultExprs.size(); ++i) {
             SlotRef slotRef = new SlotRef(tupleDesc.getSlots().get(i));
             resultExprs.set(i, slotRef);
-            selectStmt.getBaseTblResultExprs().set(i, slotRef);
```
Comment (imay) [inline fe/src/main/java/org/apache/doris/planner/SingleNodePlanner.java]: Is it OK when executing `insert into` statement?
I remember it seems to be introduced here in support of `insert into values()`. You can try if this patch works properly.

**Review by imay — CHANGES_REQUESTED** [review]
License Header

File: fe/src/main/java/org/apache/doris/planner/SingleNodePlanner.java (hunk 2 above)
Comment (kangkaisen) [reply fe/src/main/java/org/apache/doris/planner/SingleNodePlanner.java]: Has reverted this change.

**Review by imay — APPROVED** [review]
LGTM

**Pull request merged**